command/ca/provisionerbeta/add.go

```diff
@@ -0,0 +1,702 @@
+'''
+
+Create an Azure provisioner with two service groups:
+'''
+$ step beta ca provisioner add Azure --type Azure \
+  --azure-tenant bc9043e2-b645-4c1c-a87a-78f8644bfe57 \
+  --azure-resource-group identity --azure-resource-group accounting
+'''
+
+Create an GCP provisioner that will only accept the SANs provided in the identity token:
+'''
+$ step beta ca provisioner add Google --type GCP \
+  --disable-custom-sans --gcp-project internal
+'''
+
+Create an AWS provisioner that will only accept the SANs provided in the identity
+document and will allow multiple certificates from the same instance:
+'''
+$ step beta ca provisioner add Amazon --type AWS \
+  --aws-account 123456789 --disable-custom-sans --disable-trust-on-first-use
+'''
+
+Create an AWS provisioner that will use a custom certificate to validate the instance
+identity documents:
+'''
+$ step beta ca provisioner add Amazon --type AWS \
+  --aws-account 123456789 --iid-roots $(step path)/certs/aws.crt
+'''`,
+	}
+}
+
+func addAction(ctx *cli.Context) (err error) {
+	if err := errs.NumberOfArguments(ctx, 1); err != nil {
+		return err
+	}
+
+	x509TemplateFile := ctx.String("x509-template")
+	x509TemplateDataFile := ctx.String("x509-template-data")
+	sshTemplateFile := ctx.String("ssh-template")
+	sshTemplateDataFile := ctx.String("ssh-template-data")
+
+	args := ctx.Args()
+
+	typ := ctx.String("type")
+
+	p := &linkedca.Provisioner{
+		Name: args.Get(0),
+	}
+
+	// Read x509 template if passed
+	p.X509Template = &linkedca.Template{}
+	if x509TemplateFile != "" {
+		b, err := utils.ReadFile(x509TemplateFile)
+		if err != nil {
+			return err
+		}
+		p.X509Template.Template = b
+	}
+	if x509TemplateDataFile != "" {
+		b, err := utils.ReadFile(x509TemplateDataFile)
+		if err != nil {
+			return err
+		}
+		p.X509Template.Data = b
+	}
+	// Read ssh template if passed
+	p.SshTemplate = &linkedca.Template{}
+	if sshTemplateFile != "" {
+		b, err := utils.ReadFile(sshTemplateFile)
+		if err != nil {
+			return err
+		}
+		p.SshTemplate.Template = b
+	}
+	if sshTemplateDataFile != "" {
+		b, err := utils.ReadFile(sshTemplateDataFile)
+		if err != nil {
+			return err
+		}
+		p.SshTemplate.Data = b
+	}
+
+	p.Claims = &linkedca.Claims{
+		X509: &linkedca.X509Claims{
+			Durations: &linkedca.Durations{
+				Min:     ctx.String("x509-min-dur"),
+				Max:     ctx.String("x509-max-dur"),
+				Default: ctx.String("x509-default-dur"),
+			},
+			Enabled: !(ctx.IsSet("x509") && !ctx.Bool("x509")),
+		},
+		Ssh: &linkedca.SSHClaims{
+			UserDurations: &linkedca.Durations{
+				Min:     ctx.String("ssh-user-min-dur"),
+				Max:     ctx.String("ssh-user-max-dur"),
+				Default: ctx.String("ssh-user-default-dur"),
+			},
+			HostDurations: &linkedca.Durations{
+				Min:     ctx.String("ssh-host-min-dur"),
+				Max:     ctx.String("ssh-host-max-dur"),
+				Default: ctx.String("ssh-host-default-dur"),
+			},
+			Enabled: !(ctx.IsSet("ssh") && !ctx.Bool("ssh")),
```

Comment:
These expressions are confusing, basically, both x509 and ssh will be enabled unless you --x509=false or --ssh=false. Is that intended?

And I don't think we currently support x509.enabled=false

Reply:
So what do you think should be the default? ssh enabled or disabled?

Reply:
I figured it would be easier and more natural if it was just auto enabled and then you had to turn it off. I can remove the existing flags and just have a `--disable-ssh`.

Reply:
I guess it's ok, before we were doing `--ssh` and I think it was a little bit confusing for our users.